packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx

```diff
@@ -0,0 +1,174 @@
+import {Hidden} from '../../../common/hidden';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+import ArrowBottomIcon from 'coveo-styleguide/resources/icons/svg/arrow-bottom-rounded.svg';
+import {InitPopoverEventPayload} from './popover-event';
+
+/**
+ * @internal
+ * The `atomic-popover` displays a popover menu view of any facet
+ * @part value-button - The button used to display the popover menu
+ * @part value-label - The popover value label
+ * @part arrow-icon - The arrow icon
+ * @part placeholder - The placeholder displayed when the facet isn't ready
+ * @part popover-wrapper - The wrapper that contains the value-button and the slot-wrapper
+ * @part slot-wrapper - The wrapper that contains the facet/slot
+ */
+
+@Component({
+  tag: 'atomic-popover',
+  styleUrl: 'atomic-popover.pcss',
+  shadow: true,
+})
+export class AtomicPopover implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @State()
+  public error!: Error;
+
+  @State() public isMenuVisible = false;
+  @State() public facetId?: string;
+  @State() public facetLabel?: string = 'no-label';
+  @Element() host!: HTMLElement;
+  private facetElement?: HTMLElement;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  popoverOpened() {
+    if (!this.isMenuVisible) {
+      const popoverOpened = new CustomEvent('popoverOpened', {
+        detail: this.facetId,
+      });
+      document.dispatchEvent(popoverOpened);
+    }
+
+    this.isMenuVisible = !this.isMenuVisible;
+  }
+
+  @Listen('popoverOpened', {target: 'document'})
+  popOpened(event: CustomEvent) {
+    if (event.detail !== this.facetId) {
+      this.isMenuVisible = false;
+    }
+  }
+
+  renderValueButton() {
+    return (
+      <Button
+        style="square-neutral"
+        onClick={() => this.popoverOpened()}
+        part="value-button"
+        class={`value-button rounded flex box-border h-full items-center mr-1.5 p-2.5 group ${
+          this.isMenuVisible
+            ? 'selected border-primary shadow-outer-light-blue text-primary'
+            : 'hover:border-primary-light focus-visible:border-primary-light '
+        }`}
+      >
+        <span
+          title={this.facetLabel}
+          part="value-label"
+          class={`value-label truncate ${
+            this.isMenuVisible
+              ? ''
+              : 'group-hover:text-primary-light group-focus:text-primary'
+          }`}
+        >
+          {this.facetLabel}
+        </span>
+        <atomic-icon
+          part="arrow-icon"
+          class={`w-2 ml-2 ${
+            this.isMenuVisible
+              ? 'rotate-180'
+              : 'group-hover:text-primary-light group-focus:text-primary'
+          } `}
+          icon={ArrowBottomIcon}
```

Comment:
I would add `aria-hidden="true` to the icon to remove it from screen readers' accessibility DOM, since it's adequately described by the button.